WIP: Add 'interfaces' service template

ndom91 opened the pull request:
Add `interfaces` service template to ease the pulling of interface details with the end goal of creating a csv which one can easily import into the [Netbox](https://github.com/netbox-community/netbox) bulk interface importer. 

Current idea is to start with very basics and maybe just wrap this in a little CLI which can simply output the csv which one can paste into netbox. Further enhancements could include directly calling the netbox api to add said interfaces or something like that.

#### Notes

Netbox interface importer example: https://netboxdemo.com/dcim/interfaces/import
> Login: **netbox**/**netbox**

Fields:
```
device, name, type, enabled, mac_address, mtu, description, mode 
```
- device - device name which the interface belongs to
- name - interface name
- type - physical medium (SFP, 10/100, 1G, etc.)
- enabled - boolean on/off
- mac_address - mac address
- mtu - mtu size
- description - description
- mode - 802.1Q Mode (access, tagged, tagged (all))

tbotnz commented on backend/plugins/extensibles/j2_service_templates/interfaces.j2:
```diff
@@ -0,0 +1,69 @@
+					}
+        }
+      },
+      {
```
please remove this object as its not required

tbotnz commented on backend/plugins/extensibles/j2_service_templates/interfaces.j2:
```diff
@@ -0,0 +1,69 @@
+  {% for host in hosts %}
+  {
+    "supported_methods": [
+      {
```
please remove this object as its not required

tbotnz commented on backend/plugins/extensibles/j2_service_templates/interfaces_model.py:
```diff
@@ -0,0 +1,32 @@
+        schema_extra = {
+            "example": {
+                "operation": "retrieve",
+                "args":{
```
the args should also include driver & command

ndom91 replied:
Whoops forgot to commit that update. Its in now.

ndom91 commented:
Updated the name and cleaned up the unnecessary objects and what not.

I accidentally commited my local copy of the docker-compose and config.json files accidentally and then deleted them. I hope merging this wont delete them completely. If so, I'll have to see if I can revert the last two commits and redo that correctly :sweat_smile:

tbotnz commented on backend/plugins/extensibles/j2_service_templates/netmiko_retrieve_data_model.py:
```diff
@@ -0,0 +1,34 @@
+    driver: str
+    command: str
+
+class interface_model(BaseModel):
```
plz rename this netmiko_retrieve_data_model

tbotnz commented on backend/plugins/extensibles/j2_service_templates/netmiko_retrieve_data_model.py:
```diff
@@ -0,0 +1,34 @@
+from backend.core.models.service import service_lifecycle
+from backend.core.models.models import queue_strat
+
+class interface_model_args(BaseModel):
```
plz rename this netmiko_retrieve_data_model_args

tbotnz commented on backend/plugins/extensibles/j2_service_templates/netmiko_retrieve_data_model.py:
```diff
@@ -0,0 +1,34 @@
+class interface_model(BaseModel):
+    # this class MUST match the filename & the filename must be formatted $servicetemplatename_model.py
+    operation: service_lifecycle
+    args: interface_model_args
```
plz rename this netmiko_retrieve_data_model_args

tbotnz commented on backend/plugins/extensibles/j2_service_templates/netmiko_retrieve_data.j2:
```diff
@@ -0,0 +1,39 @@
+{# NETPALM SERVICE BOILERPLATE #}
```
the comments can be removed {# #}

ndom91 commented:
Okay, should be cleaned up then :+1:

ndom91 commented:
For some reason I can't get this to work in the web openapi docs although it was working fine in my postman setup in the office. 

I now get the error:
```
 "validate_template Expecting property name enclosed in double quotes: line 13 column 6 (char 276)",
      "render_service 'NoneType' object is not iterable"
```
returned when I check the task

tbotnz commented on config.json:
```diff
@@ -1,40 +0,0 @@
-{
```
why remove?

tbotnz commented on docker-compose.yml:
```diff
@@ -1,56 +0,0 @@
-version: "3.7"
```
why remove?

ndom91 replied:
Yeah didn't mean to. I had accidentally commit my local copies that had the volumes mounted, in docker-compose and my own api-key in config.json, then when I tried to remove my accidental commit from this PR it lead to this mess.. Haha.

tbotnz replied:
No worries, just delete the two files from your PR and ill merge it

ndom91 replied:
Okay so I restored those two files back to the state of the upstream that I forked from, so I'm not sure why theres those few small differences in there now.
